### PR TITLE
Fix lib to work with error fields and post requests 

### PR DESCRIPTION
**Pull request by AleksandrShepelev**
1. there was no error model in all api responses
2. signature to do post requests require request body to be added to signature, so refactoring of internal code was done (no external lib api changes)
3. Amount and total in request model now made pointers because otherwise coinbase api said that both numbers specified

### CODE REVIEW

**Comment by AleksandrShepelev** [comment]
done. In addition, I've created separate model for requests with name field and moved it to global file.

File: test/prices_test.go [hunk 1]
```diff
@@ -61,8 +61,8 @@ func TestGetSpotPrice(t *testing.T) {
   if balance.Data.Currency != "EUR" {
     t.Error("Expected EUR, got ", balance.Data.Currency)
   }
-  if balance.Data.Amount < 900.0 {
-    t.Error("Expected amount >= 900, got ", balance.Data.Amount)
+  if balance.Data.Amount < 500.0 {
```
Comment (Zauberstuhl) [inline test/prices_test.go]: That test clearly doesn't work as expected. Would you change it into something like this:

```
  if balance.Data.Amount <= 0.0 {
    t.Error("Expected amount >= 0.0, got ", balance.Data.Amount)
  }
```

If you [squash](https://github.com/todotxt/todo.txt-android/wiki/Squash-All-Commits-Related-to-a-Single-Issue-into-a-Single-Commit) your commits I am happy to merge :+1:

Comment (AleksandrShepelev) [reply test/prices_test.go]: image
you can squash commits during merge 
but if you want, I may squash commits by myself

**Comment by Zauberstuhl** [comment]
Awesome :tada: Thanks for your work

**Pull request merged**